crates/telio-utils/src/dual_target.rs

```diff
@@ -14,6 +14,29 @@ pub type Result<T> = std::result::Result<T, DualTargetError>;
 /// Target input with IPv4 and IPv6 addresses
 pub type Target = (Option<Ipv4Addr>, Option<Ipv6Addr>);
 
+/// Builds Target from a list of ip addresses
+pub fn build_target(ip_addresses: &Vec<IpAddr>, use_ipv6: bool) -> Target {
+    let mut ipv4 = None;
+    let mut ipv6 = None;
+
+    for ip in ip_addresses {
+        match ip {
+            IpAddr::V4(v4) => {
+                if ipv4.is_none() {
```

Comment:
NIT: you could avoid additional identation and merge if with mach arm. adding default as none case